control-plane/csi-driver/src/bin/node/mount.rs

```diff
@@ -297,3 +297,56 @@ pub(crate) fn blockdevice_unmount(target: &str) -> Result<(), Error> {
     info!("block device at {} has been unmounted", target);
     Ok(())
 }
+
+/// Waits until a device's filesystem is shutdown.
+/// This is useful to know if it's safe to detach a device from a node or not as it seems that
+/// even after a umount completes the filesystem and more specifically the filesystem's journal
+/// might not be completely shutdown.
+/// Specifically, this waits for the filesystem (eg: ext4) shutdown and the filesystem's journal
+/// shutdown: jbd2.
+pub(crate) async fn wait_fs_shutdown(device: &str, fstype: Option<String>) -> Result<(), Error> {
+    let device_trim = device.replace("/dev/", "");
+
+    let start = std::time::Instant::now();
```

Comment:
Do we intentionally use one start time in all subsequent 'wait_file_removal' rather than independent start time per 'wait_file_removal' ?
So the semantics of this single start time is 'all wait operations should not last longer than X' ?

Reply:
yeah, I thought 2s would be more than enough, otherwise let kubernetes retry later.